File: testapp/src/main/java/com/juanimoli/strictmodenotifier/testapp/LeakedSQLiteOpenHelper.java
package com.juanimoli.strictmodenotifier.testapp;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class LeakedSQLiteOpenHelper extends SQLiteOpenHelper {

    static final String DB = "sqlite_sample.db";
    static final int DB_VERSION = 1;
    static final String CREATE_TABLE =
            "create table sample ( _id integer primary key autoincrement, data integer not null );";
    static final String DROP_TABLE = "drop table sample;";

    public LeakedSQLiteOpenHelper(Context context) {
        super(context, DB, null, DB_VERSION);
    }

    public void onCreate(SQLiteDatabase db) {
        db.execSQL(CREATE_TABLE);
    }

    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(DROP_TABLE);
        onCreate(db);
    }
}

File: testapp/src/main/java/io/github/juanimoli/strictmode/notifier/helper/LeakedSQLiteOpenHelper.kt
package io.github.juanimoli.strictmode.notifier.helper

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class LeakedSQLiteOpenHelper(context: Context?) : SQLiteOpenHelper(context, DB, null, DB_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(CREATE_TABLE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL(DROP_TABLE)
        onCreate(db)
    }

    companion object {
        private const val DB = "sqlite_sample.db"
        private const val DB_VERSION = 1
        private const val CREATE_TABLE =
            "CREATE TABLE sample (_id INTEGER PRIMARY KEY AUTOINCREMENT, data INTEGER NOT NULL );"
        private const val DROP_TABLE = "DROP TABLE sample;"
    }
}
